### DISCUSSION TITLE
[@types/better-sqlite3] Class properties of `SqliteError` are inaccessible.

### DISCUSSION THREAD

**Discussion by atifcppprogrammer** [DefinitelyTyped/DefinitelyTyped#71131] [Issues with a @types package]
Hi there ! it seems that the `SqliteError` class has been typed incorrectly since the use of `type SqliteError = typeof SqliteError` line at the following two instances

- https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/better-sqlite3/index.d.ts#L91
- https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/better-sqlite3/index.d.ts#L141 

prevents access to the class's member properties i.e. `name`, `message` and `code`. The `code` property is particularly important since it corresponds to SQLite3 error codes as per the package documentation [here](https://github.com/WiseLibs/better-sqlite3/blob/master/docs/api.md#class-sqliteerror), having this information is crucial for my use case.

[a comment by an automated account is not shown]

**Comment by beenotung** [comment]
@atifcppprogrammer I can get the class property of `SqliteError` with below setup. See if we're having different setup

test.ts:
```typescript
import DB, { SqliteError } from 'better-sqlite3'

let db = new DB('test.sqlite3')

try {
    db.exec('create table user ( ')
} catch (error) {
    if (error instanceof SqliteError) {
        console.log('sqlite error:', error)

        // this doesn't cause typescript error, and it gets "SQLITE_ERROR"
        console.log('error code:', error.code)

        // this gets "SqliteError"
        console.log('error code:', error.name)
    } else {
        console.log('unknown error:', error)
    }
}
```

tsconfig.json:
```json
{
    "compilerOptions": {
        "target": "es2022",
        "module": "commonjs",
        "esModuleInterop": true,
        "strict": true,
        "skipLibCheck": true
    }
}
```


package.json:
```json
{
    "devDependencies": {
        "@types/better-sqlite3": "^7.6.9999",
        "@types/node": "^22.9.9999",
        "ts-node": "^4.1.0",
        "typescript": "^5.6.3"
    },
    "dependencies": {
        "better-sqlite3": "^11.5.0"
    }
}
```
